### DISCUSSION TITLE
Modern motes: gecko, nrf52840, and nrf

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2472] [Q&A] [answered]
I am building a testbed and thought it would be a good idea to use the most powerful available for longevity (I'm planning on donating the testbed at the end of my research). Based on my research, those will be the `gecko` and `nrf` platforms.

1. I don't see `gecko` as a target. Was it added after v4.7?
2. What is the difference between using `nrf52840` vs. `nrf` targets?
3. The `PCA10095 (nRF5340-DK)` has 512 KB of RAM. Do we have access to all of it or only half (it is dual core)?
4. It also has a 64 MB external flash memory. Do we have access to it from Contiki?

**Comment by pjonsson** [comment, marked as answer]
I can answer half of this, so I'll do that.

1. Yes, and 4.9 is right around the corner.
2. The nrf target supports nrf 5340, and is missing support for something with the 52840. The long term plan is to only have the nrf target, but the missing support from the nrf target will not get done to 4.9.

**Reply by rahav** [reply]
2 x thank you @pjonsson for the answers. Appropose 4.9, are these two platforms the newer of the bunch? Or is there another one you suggest I consider?

The requirements for the testbed are: Contiki is well supported and stable on the platform (predominantly for STEM education to not run into challenges later on), IPv6 on 802.15.4, and lots of RAM and ROM.
